### PR TITLE
feat: add number comparison node

### PR DESCRIPTION
**Pull request by Le-Pepp**

### CODE REVIEW

File: basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/NumberOrderingNodeType.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    @Override
+    public Map<String, DataBox<?>> settings() {
+        return Map.of("mode", NUMBER_ORDERING_MODE.create(Mode.GREATERTHAN));
+    }
+
+    @Override
+    public Set<NodeConnector.Input<?>> inputs(UUID nodeId, Map<String, DataBox<?>> settings) {
+        return Set.of(
+            new NodeConnector.Input<>(nodeId, "inputA", DataType.NUMBER),
+            new NodeConnector.Input<>(nodeId, "inputB", DataType.NUMBER)
```
Comment (williambl) [inline basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/NumberOrderingNodeType.java:26]: These can probably just be named 'a' and 'b'?

File: basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/NumberOrderingNodeType.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import club.mondaylunch.gatos.core.data.DataBox;
+import club.mondaylunch.gatos.core.data.DataType;
+import club.mondaylunch.gatos.core.graph.connector.NodeConnector;
+import club.mondaylunch.gatos.core.graph.type.NodeType;
+
+public class NumberOrderingNodeType extends NodeType.Process {
```
Comment (williambl) [inline basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/NumberOrderingNodeType.java:13]: maybe rename this number comparison? number ordering made me think of list sorting or smth at first.

**Review by williambl — COMMENTED** [review]
code-wise looks perfectly fine, just a couple of naming things.

**Comment by williambl** [comment]
also you need to rebase/merge from master :D

File: basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/NumberOrderingNodeType.java (hunk 1 above)
Comment (Le-Pepp) [reply basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/NumberOrderingNodeType.java:26]: Convention with other nodes has been "input/output<letter>". If, for whatever reason, this node had multiple outputs the distinction would be useful to preserve.

File: basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/NumberOrderingNodeType.java (hunk 2 above)
Comment (Le-Pepp) [reply basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/NumberOrderingNodeType.java:13]: blame paul :P

**Review by williambl — APPROVED** [review]

**Comment by fenn7** [comment]
Resolve conversing then this can be merged

**Pull request merged**